src/extensions/agent_modules/shared/mosquitto_utils/src/adu_mosquitto_utils.c

```diff
@@ -140,7 +146,25 @@ bool ADU_are_correlation_ids_matching(
 
     if (outCorrelationData != NULL)
     {
-        if (ADUC_AllocAndStrCopyN(outCorrelationData, responseCorrelationId, (size_t)responseCorrelationIdLen) != 0)
+        // TODO: Remove this workaround once adu service deploys fix that ensures exact correlation id string agent sent is
```

Comment:
is there a bug tracking the service side fix?

Reply:
please add the # to the comment.

Reply:
Here it is @Jerry-GeGarcia  and @JeffMill : [Bug 48216724](https://microsoft.visualstudio.com/OS/_workitems/edit/48216724): [gen2] service returns modified correlation id in the response